### ISSUE TITLE
Contents of acn-sdk-c are not available for working with Cypress PSoC

### ISSUE THREAD

**Issue by shreyas2415** [arrow-acs/acn-embedded#10]
I just cloned master branch into my git and i am not able to see the contents of acn-sdk-c. So when I open the project in Cypress  I have all question marks. 
![capture_1](https://user-images.githubusercontent.com/20415525/37645019-ebc91be4-2c25-11e8-8ab7-fd40104a1d10.PNG)
Is there a specific branch which I need to use? The Master branch is broken

**Comment by DanilaDemidov** [comment]
Hi,
I have updated the SenseAblitity project. There were some file links don't exist anymore. But in your case it seems there are no any files into acn-sdk-c submodule.
Please use --recursive flag when you clone this repository.
And I should notice: it was just a example. There is no socket layer implementation only dummy function. Also this platform have no much memory by default that's why I used these defines for the private.h file:
#define NO_EVENTS
#define NO_RELEASE_UPDATE
#define NO_SOFTWARE_UPDATE

#define DBG_LINE_SIZE           32
#define MQTT_BUF_LEN            300
#define RINGBUFFER_SIZE         256

**Comment by shreyas2415** [comment]
Thanks ... I am just evaluating the BLE solution implemented. I may not be really checking Socket stuff. So it is okay. I wanted to check BLE aspect. I am still seeing loads on "include" file issues. I think the relative paths aren't being picked up. Once I am done, may be I can share the files with you for checking in! :) Or I can do it in my branch and then you can merge to main line

**Comment by DanilaDemidov** [comment]
I think it's highly possible. And there are some unnecessary files in this project. I want to clean it. Please send me patch file or maybe pull request. Any help are welcome. Thanks!

**Comment by shreyas2415** [comment]
Okay i will do the same.. I will try it out on my PSoC4200 and will send the changes across as soon as the compilation works out. Thanks so much

**Comment by shreyas2415** [comment]
![capture](https://user-images.githubusercontent.com/20415525/37662014-6800e70c-2c56-11e8-9116-4e9f5d7774e1.PNG)
I am currently stuck here compiling. May be I have to define the flags correctly. I am using PSoC 4200 and I am working on Windows 10. Rest of the issues I have fixed. Could you please check and tell me

**Comment by dandemidow** [comment]
I have not this issue but I suppose that you got this error trying to build the wolfSSL. Do you really need it?
Anyway check your compiler flags, here should be -DARCH_SSL flag. This flag includes acnsdkc/sys/arch/ssl.h file.

**Comment by shreyas2415** [comment]
Okay sure i will do that and give it a shot tomorrow. I really don't need a wolfSSL for now.. Thanks ! :)

**Comment by DanilaDemidov** [comment]
I have cleaned all useless files from the project.

**Comment by shreyas2415** [comment]
Okay thanks.. i will merge your changes in my branch and continue from there ? :) thanks

**Comment by DanilaDemidov** [comment]
I believe you can do this. I just remove temporary files which generated by the PSoC Creator and add these into .gitignore.

**Comment by shreyas2415** [comment]
Hi, Pulled in the latest code, added include directories in my Cypress project and I am still unable to compile. Does anyone have a working copy / support for Cypress project for this ? If yes, please send it out. 
![capture1](https://user-images.githubusercontent.com/20415525/37910972-26fd85c2-310f-11e8-9d4e-fee4d96676ca.PNG)

**Comment by DanilaDemidov** [comment]
[SenseAbility-000.cywrk.Archive01.zip](https://github.com/arrow-acs/acn-embedded/files/1851561/SenseAbility-000.cywrk.Archive01.zip)
I have packed my project through PSoC Creator. Your issue is very weird. Can you share your test project? I can take a look.

**Comment by shreyas2415** [comment]
https://drive.google.com/open?id=161RrKrLCgDcrfZfqG7j2h5nytd6w2ERN 
I have shared my project here and can we try this again ?

**Comment by shreyas2415** [comment]
@DanilaDemidov I tried your Zip as well. It still doesn't compile. Very strange. It says files missing. How is it compiling for you guys. Do I need to set some environment variables etc ?

**Comment by dandemidow** [comment]
@shreyas2415 I didn't set any variables. I'll take a look you project tomorrow. I hope we find the issue.

**Comment by DanilaDemidov** [comment]
@shreyas2415 Now I'm understanding your issue. I have fixed a couple of things. And added the private.h file also. The API and secret keys are very important stuff for an ACN. I placed into your private file some random numbers. You can get the real one in the www.arrowconnect.io
Send your project back with my fixes: https://drive.google.com/open?id=1-_Kyff7dLWWpn6P1gPzYVbhr6UomHhD8

**Comment by shreyas2415** [comment]
Hi @DanilaDemidov 
Thanks so much. However I am using the following version of PSoC Creator ==> 
![psoc_version](https://user-images.githubusercontent.com/20415525/38307529-4c96f2d0-3814-11e8-82a9-31b3ae51d372.PNG)
and i am still getting an error
![error](https://user-images.githubusercontent.com/20415525/38307567-65025abc-3814-11e8-82b1-b63989b44ba1.PNG)

I think it is best that you send me a .hex image so I know if it works then we can evaluate your solution ?

**Comment by DanilaDemidov** [comment]
@shreyas2415 
I see, I think there is issue with compiler parameters:
Check the Build Settings -> ARM GCC (or your compiler) -> Additional Include Directories
![image](https://user-images.githubusercontent.com/26493218/38308068-952762a0-381e-11e8-9cc6-ee630e6bdf00.png)

**Comment by shreyas2415** [comment]
Well i am using your build directly, so I have all the same directories as you have listed in the folder.. let me paste that 
![capture](https://user-images.githubusercontent.com/20415525/38308484-851a3e66-3817-11e8-8b81-a1228dcc27ea.PNG)
I prefer if you just share the hex image which i will program and see if that works for me.

**Comment by DanilaDemidov** [comment]
@shreyas2415 
Not exactly. You have **acnsdk** instead **acnsdkc**. Please fix it.
And these include path can affect your build:
acn-sdk-c/helper 
acn-sdk-c/skeleton 
acn-sdk-c/src
acnsdkc/arrow
acnsdkc/bsd
acnsdkc/json
acnsdkc/sys
acnsdkc/time
It would be better to remove these paths.

And another parameter **Preprocessor Definitions** should contain this: 

`DEBUG; __senseability__; ARCH_SSL`

**Comment by DanilaDemidov** [comment]
[Debug.zip](https://github.com/arrow-acs/acn-embedded/files/1875823/Debug.zip)
This is my hex build but I don't understand how you check it. It's just a half-finished product.

**Comment by shreyas2415** [comment]
Yeah i cannot really use your hex build for my product. But I am still unable to compile. I am still getting the errors pthread_mutex_t and sys/uio.h file not found

**Comment by DanilaDemidov** [comment]
It’s better I believe. Can you show the Preprocessor Definitions for this project?

**Comment by shreyas2415** [comment]
Only one flag is defined 
NDEBUG

**Comment by DanilaDemidov** [comment]
Please replace it
DEBUG
_ _ senseability _ _
ARCH_SSL

Remove spaces for a second define before)

**Comment by shreyas2415** [comment]
@DanilaDemidov Works finally compilation works for my device and I am able to flash it as well. I am not sure if it is actually connecting to arrow.. I will check with my colleague today/tomorrow who has worked on arrow.io and see. Thanks for the excellent support 🥇
